### ISSUE TITLE
User MemberOf deserialized as DirectoryObject

### ISSUE THREAD

**Issue by Dangles91** [microsoftgraph/msgraph-beta-sdk-dotnet#281]
When trying to use the beta SDK to retrieve a list of groups a user is a member of via

`client.Users[objectId].MemberOf.Request().GetAsync()`

The returned collection contains DirectoryObject objects rather than the expected Group objects. The @odata.type for in the raw JSON is correctly returned as '#microsoft.graph.group'. However, these objects are deserialized into DirectoryObject.

I did see one similar issue with user managers where this was resolved as expected behaviour, so it might just be that I'm missing some key information on how to handle these objects. However, I have not been able to easily convert or deserialize them into the expected Group objects.

 
[AB#9003](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/9003)

**Comment by andrueastman** [comment]
Hey @Dangles91, thanks for opening this issue. 
Any chance you can share some more information about the SDK version you are using when facing this issue?

**Comment by Dangles91** [comment]
Hi @andrueastman,

Sorry for not including this in the description. This issue occurs with SDK versions 4.0.0 and 4.0.1-preview

**Comment by schmid37** [comment]
#290 seems we run in a similar problem with MobileApp

**Comment by andrueastman** [comment]
Closing this as this is now resolved in 4.3.0-preview. You should now be able to pull off something like this.

```cs
            var responsePage = await graphServiceClient.Me.MemberOf.Request().GetAsync();
            foreach (var directoryItem in responsePage)
            {
                if (directoryItem is Group group)
                {
                    Console.WriteLine(group.DisplayName);
                }
            }
```
